internal/vshard/orchestrator/failover.go

```diff
@@ -131,12 +131,19 @@ func (f *failover) shouldBeAnalysisChecked() bool {
 
 func (f *failover) checkAndRecover(ctx context.Context, analysis *ReplicationAnalysis) {
 	logger := f.logger.With().Str("ReplicaSet", string(analysis.Set.UUID)).Logger()
-	logger.Info().Msgf("checkAndRecover: %s", analysis.String())
+	logger.Debug().Msgf("checkAndRecover: %s", analysis.String())
 
 	recvFunc, desc := f.getCheckAndRecoveryFunc(analysis.State)
 	if recvFunc == nil {
 		if desc != "" {
-			logger.Warn().Msg(desc)
+			master, err := analysis.Set.Master()
```

Comment:
It seems that you should just add a field `MasterURI` to the `ReplicaSet` and use it.

Reply:
Added new field MasterURI